bot/src/main/java/org/opendaylight/bot/jenkins/Client.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc. and others. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.opendaylight.bot.jenkins;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.offbytwo.jenkins.JenkinsServer;
+import com.offbytwo.jenkins.model.JobWithDetails;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.opendaylight.bot.BotConfiguration;
+
+/**
+ * Template of Initial Jenkins Client API.
+ *
+ * @author Prateek Chanda
+ */
+@SuppressFBWarnings({"UWF_UNWRITTEN_FIELD"})
+public class Client {
+
+    private String jobName;
+    private String login;
+    private String passwd;
+
+    public static BotConfiguration getConfiguration(String admin, String password) {
+
+        BotConfiguration configuration = new BotConfiguration();
+        configuration.jenkinsBase = URI.create("https://jenkins.opendaylight.org/");
+        configuration.jenkinsLogin = admin;
+        configuration.jenkinsPassword = password;
+        return configuration;
+    }
+
+    void buildJob() throws IOException {
+
+        URI linkname = getConfiguration(login, passwd).jenkinsBase;
+        JenkinsServer jenkninsServer = new JenkinsServer(linkname, getConfiguration(login, passwd).jenkinsLogin,
+                getConfiguration(login, passwd).jenkinsPassword);
+        // Build with Parameters
+        String parameterJob = "TestingParameters";
```

Comment:
How to pass the parameters.

- User input?
- or as a parameter for the method?
- or through bot config?

Reply:
Jenkins URL, login and password would be in `BotConfiguration` like you already have it. The URL should be set to `https://jenkins.opendaylight.org` [here](https://github.com/vorburger/opendaylight-bot/blob/master/bot/src/main/java/org/opendaylight/bot/odl/OpenDaylightConfigurer.java) and the  jenkins uid & pwd read from an environment variables with a suitable name you can make up. The job name should be an argument to a method in the jenkins.Client class.

Reply:
ok, got it.
I will update that in my next PR

Reply:
I guess the URL has already been set to `https://jenkins.opendaylight.org` ?

We just need the parameters for Job as a method parameter and uid ans pwd as a environment variable, I guess ?